### PR TITLE
[SERVICE-791] Revisit return type of subscription methods

### PR DESCRIPTION
**Pull request by bryangaleopenfin**
Modifies `addContextListener`/`addEventListener` on our ChannelObject to return synchronously, and changes `broadcast` on our Channel object to return asynchronously, to match equivalent calls in `main`

Also does some tidying of how intent handlers are invoked in demo apps

### CODE REVIEW

File: src/demo/apps/NewsApp.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export function NewsApp(): React.ReactElement {
             }
         });
 
-        const intentListener = fdc3.addIntentListener(fdc3.Intents.VIEW_NEWS, (context: Context): Promise<void> => {
+        const intentListenerPromise = fdc3.addIntentListener(fdc3.Intents.VIEW_NEWS, (context: Context): Promise<void> => {
             return new Promise((resolve, reject) => {
                 try {
                     handleIntent(context as InstrumentContext);
```
Comment (ottah) [inline src/demo/apps/NewsApp.tsx]: Side note but can't all of these listeners in the component just be:
```ts
async (context: Context) => {
	await handleIntent(context as InstrumentContext)
}
```

**Review by ottah — APPROVED** [review]
Looks good.

File: src/demo/apps/NewsApp.tsx (hunk 1 above)
Comment (bryangaleopenfin) [reply src/demo/apps/NewsApp.tsx]: Hmm...

Yeah, not sure why this is like this. Will tidy and see if anything breaks

Comment (bryangaleopenfin) [reply src/demo/apps/NewsApp.tsx]: Sorted in News, Dialer, and Charts.

**Review by ottah — APPROVED** [review]

**Review by tomrobertsOF — APPROVED** [review]

**Pull request merged**